tests/cxx/fwd_decl_print.cc

```diff
@@ -0,0 +1,114 @@
+//===--- fwd_decl_print.cc - test input file for iwyu ---------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// IWYU_ARGS: -I . -std=c++20
+
+// Test the printing of forward-declare lines for forward-declarable types.
+
+#include "tests/cxx/fwd_decl_print-d1.h"
+#include "tests/cxx/fwd_decl_print-d2.h"
+
+// Simple tag types.
+Enum* e1;
+Class* c1;
+Struct* s1;
+
+ns1::EnumInNamespace* e2;
+ns1::ClassInNamespace* c2;
+ns1::StructInNamespace* s2;
+
+// Derived class types.
+DerivedClass* dc1;
+
+// Class types with explicit and implicit attributes (attrs should not be
+// included in fwd decl).
+ClassWithStdAttr* ac1;
+ClassWithGNUAttr* ac2;
+ClassWithAlignAs* ac3;
+ClassWithFinal* ac4;
+
+// Simple templates.
+Template<int>* t1;
+ns1::TemplateInNamespace<char>* t2;
+
+// Derived templates
+DerivedTemplate<int>* dt1;
+
+// Templates with explicit and implicit attributes (attrs should not be included
+// in fwd decl).
+TemplateWithStdAttr<int>* at1;
+TemplateWithGNUAttr<int>* at2;
+// TODO: this results in:
+//     template <class T> class alignas(8)  TemplateWithAlignAs;
+// but the alignas attribute should not be included.
```

Comment:
They are acceptable in forward-declarations, they should just be the same if present.

Reply:
Updated the commentary to make that clearer. I might follow up with a later fix for `alignas` consistency.